src/Rtp/ExtendedSequenceCounter.cpp

```diff
@@ -39,77 +43,131 @@ bool ExtendedSequenceCounter::UpdateState(rtp_sequence_num_t seq)
+            .extendedSeq = cycles | seq,
+            .valid = true,
+            .resync = false,
+            .stable = true,
+        };
+    }
+    else if (udelta <= RTP_SEQ_MOD - MAX_MISORDER)
+    {
         /* the sequence number made a very large jump */
-        if (seq == badSeq) {
+        if (seq == badSeq)
+        {
             /*
-            * Two sequential packets -- assume that the other side
-            * restarted without telling us so just re-sync
-            * (i.e., pretend this was the first packet).
-            */
-            Reset(seq);
+             * Two sequential packets -- assume that the other side
+             * restarted without telling us so just re-sync
+             * (i.e., pretend this was the first packet).
+             */
+            spdlog::trace("Sequence counter reset");
+            resync(seq);
+            return ExtendResult{
+                .extendedSeq = cycles | seq,
+                .valid = false,
+                .resync = true,
+                .stable = false,
+            };
         }
-        else {
-            badSeq = (seq + 1) & (RTP_SEQ_MOD-1);
-            return false;
+        else
+        {
+            badSeq = (seq + 1) & (RTP_SEQ_MOD - 1);
+            return ExtendResult{
+                .extendedSeq = cycles | seq,
+                .valid = false,
+                .resync = false,
+                .stable = true,
+            };
         }
-    } else {
+    }
+    else
+    {
         /* duplicate or reordered packet */
+        return ExtendResult{
+            .extendedSeq = cycles | seq,
+            .valid = true,
+            .resync = false,
+            .stable = true,
+        };
     }
-    received++;
-    return true;
 }
 
-void ExtendedSequenceCounter::Initialize(rtp_sequence_num_t seq)
+/**
+ * Per RFC 3550:
+ * When a new source is heard for the first time, that is, its SSRC
+ * identifier is not in the table (see Section 8.2), and the per-source
+ * state is allocated for it, s->probation is set to the number of
+ * sequential packets required before declaring a source valid
+ * (parameter MIN_SEQUENTIAL) and other variables are initialized:
+ * 
+ *    init_seq(s, seq);
+ *    s->max_seq = seq - 1;
+ *    s->probation = MIN_SEQUENTIAL;
+ */
+void ExtendedSequenceCounter::initialize(rtp_sequence_num_t seq)
 {
-    Reset(seq);
+    reset(seq);
     baseSeq = seq;
     maxSeq = seq - 1;
     initialized = true;
 }
 
-void ExtendedSequenceCounter::Reset(rtp_sequence_num_t seq)
+void ExtendedSequenceCounter::reset(rtp_sequence_num_t seq)
 {
     baseSeq = seq;
     maxSeq = seq;
-    badSeq = RTP_SEQ_MOD + 1;   /* so seq == badSeq is false */
+    badSeq = RTP_SEQ_MOD + 1; /* so seq == badSeq is false */
     cycles = 0;
-    received = 0;
     receivedPrior = 0;
     expectedPrior = 0;
 }
 
-std::ostream& operator<<(std::ostream & os, const ExtendedSequenceCounter& c)
+void ExtendedSequenceCounter::resync(rtp_sequence_num_t seq)
```

Comment:
This naming obscures finding all calls to `reset`. Recommend removing unless you expect `resync` to diverge soon.

Reply:
Fixed, thanks for the feedback!